editor/components/CrashDiagramControls.tsx

```diff
@@ -0,0 +1,145 @@
+import { Dispatch, SetStateAction } from "react";
+import Form from "react-bootstrap/Form";
+import Button from "react-bootstrap/Button";
+import ButtonGroup from "react-bootstrap/ButtonGroup";
+import {
+  UseFormRegister,
+  UseFormSetValue,
+  UseFormHandleSubmit,
+  SubmitHandler,
+} from "react-hook-form";
+import {
+  FaMagnifyingGlassPlus,
+  FaMagnifyingGlassMinus,
+  FaFloppyDisk,
+} from "react-icons/fa6";
+import { FaCheckCircle } from "react-icons/fa";
+import { SlActionUndo } from "react-icons/sl";
+import { useControls } from "react-zoom-pan-pinch";
+import { CrashDiagramOrientation } from "@/types/crashDiagramOrientation";
+import AlignedLabel from "@/components/AlignedLabel";
+import PermissionsRequired from "@/components/PermissionsRequired";
+
+const allowedUserSaveDiagramRoles = ["vz-admin", "editor"];
+
+export const RotateControls = ({
+  setValue,
+  register,
+}: {
+  setValue: UseFormSetValue<CrashDiagramOrientation>;
+  register: UseFormRegister<CrashDiagramOrientation>;
+}) => {
+  return (
+    <div className="mt-2">
+      <Form.Range
+        {...register("rotation")}
+        min="-180"
+        max="180"
+        id="formControlRange"
+        onChange={(e) =>
+          setValue("rotation", Number(e.target.value), { shouldDirty: true })
+        }
+        title="Rotate Diagram"
+      />
+    </div>
+  );
+};
+
+export const ZoomResetSaveControls = ({
+  setValue,
+  resetZoomToImage,
+  isDirty,
+  onSave,
+  handleSubmit,
+  isSaved,
+  setIsTouched,
+}: {
+  setValue: UseFormSetValue<CrashDiagramOrientation>;
+  resetZoomToImage: () => void;
+  isDirty: boolean;
+  onSave: SubmitHandler<CrashDiagramOrientation>;
+  handleSubmit: UseFormHandleSubmit<CrashDiagramOrientation>;
+  isSaved: boolean;
+  setIsTouched: Dispatch<SetStateAction<boolean>>;
+}) => {
+  const { zoomIn, zoomOut } = useControls();
+
+  const handleReset = () => {
+    // reset rotation
+    setValue("rotation", 0);
+    // wait for rotation to be reset before resetting zoom, to avoid trying to zoom while image is transforming
+    setTimeout(() => {
+      resetZoomToImage();
+    }, 10);
+  };
+
+  return (
+    <div className="d-flex justify-content-between w-100 mb-1">
+      <ButtonGroup>
+        <Button
+          size="sm"
+          variant="outline-primary"
+          onClick={() => {
+            zoomIn(0.25);
+            setIsTouched(true);
+          }}
```

Comment:
This is a small detail and I suspect most will use the zoom buttons, but, in the test instructions it notes to test using trackpad or scroll wheel. When I use my trackpad to zoom in or out, I don't see the save button enabled to capture the new zoom state.